### ISSUE TITLE
3.3a questions and comments

### ISSUE THREAD

**Issue by nicm** [tmux/tmux#3218]
### Ask or discuss any quick questions about 3.3 here please.

**Do not use this issue for bug reports,** please open a new issue with full information, see [CONTRIBUTING](https://github.com/tmux/tmux/blob/master/.github/CONTRIBUTING.md).

3.3a is a bug fix release of 3.3 with a few minor fixes.

**[The 3.3a CHANGES file is here](https://raw.githubusercontent.com/tmux/tmux/3.3a/CHANGES).**

As always, thanks to the many people who contributed to both this and to previous releases.

---

Here are some of the notable changes for the 3.3 release, for more see the CHANGES file:

- When building on macOS, `--enable-utf8proc` or `--disable-utf8proc` must be explicitly given to `configure` (the former is recommended).

- The default value for `default-terminal` is now chosen by `configure` from those available on the build system (force with `--with-TERM=tmux-256color`).

- There is now an ACL list to control users permitted to attach to a tmux socket (file system permissions must still be set manually); see the `server-access` command.

- The passthrough escape sequence is now controlled by an option (`allow-passthrough`), default is off (recommended).

- Various other new options: `remain-on-exit-format`, `fill-character`, `scroll-on-clear`, `pane-border-indicators`, `cursor-style`, `cursor-colour`, `popup-border-lines`, `popup-border-style`, `popup-style`, `pane-colours`.

- On Linux, tmux now supports getting the socket from systemd (socket activation), if built with `--enable-systemd`.

- There is now a menu when button 3 is pressed outside a popup border with a few items including the ability to convert a popup into a pane.

**Comment by sbbddz** [comment]
Still having problems on macOS with latest tmux. Alredy installed tmux-256color to system but still doesn't work in my terminal. Executing tmux automatically closes itself, maybe has something to do with this new version? Do someone also face this problem?

Putting screen as default terminal also does not do the job.

For now I'm sticking with tmux 3.2a, works great for me.

**Comment by nicm** [comment]
No, you will need to use a temporary file, or to extend `display-panes` to have a `-p` option.

The behaviour of passing unknown keys through is intentional, see 87ea14328ce60fdfb8bd0aef15b7e3b3bdccb1ed. I don't mind a flag to suppress them, but note that with the default `display-panes-time` and `escape-time` it is nearly impossible to press Escape and have it processed before `display-panes` times out, so you will need to either change these options or handle the Escape in your script.

Of course, a `-p` option would make a flag unnecessary.

With `-N` you will get all keys, that is the point of it.

**Comment by jemueller** [comment]
i just ran brew upgrade and it broke my tmux... you shouldnt push updates that break peoples system

**Comment by systemmonkey42** [comment]
> With `-N` you will get all keys, that is the point of it.

I settled on the following, please let me know if I'm making things hard for myself.

- Call `tmux display-panes -d0 -N`
- Read STDIN (always, don't need to treat <kbd>ESC</kbd> as an exception this way)
- Convert to key to pane with `tmux list-panes -F '#{pane_id}' -f '#{==:#{pane_index},$KEY}'

An invalid "KEY" will result in an empty "pane_id" which I can work with.

This is part of a VIM macro, and VIM is trying to parse the ESC character before I see
it, making handling it harder than necessary.

Thanks for your help.

**Comment by YellowOnion** [comment]
I'm having some weird issues where my update-environment is broken, I have a `-h` inside that won't go away, none of the WAYLAND/SWAY/XDG vars are showing:

```
> tmux show-environment
--h
DISPLAY=:1
-KRB5CCNAME
-SSH_AGENT_PID
SSH_ASKPASS=/nix/store/1axi40paycbcrn42gdb6wqw20765mf9b-seahorse-42.0/libexec/seahorse/ssh-askpass
-SSH_AUTH_SOCK
-SSH_CONNECTION
-SWAYSOCK
-WAYLAND_DISPLAY
-WINDOWID
-XAUTHORITY
-XDG_CURRENT_DESKTOP

--
> tmux show-environment -g | grep -E '(SWAY|XDG|WAYLAND)'
MOZ_ENABLE_WAYLAND=1
QT_WAYLAND_DISABLE_WINDOWDECORATION=1
SWAYSOCK=/run/user/1000/sway-ipc.1000.1703.sock
WAYLAND_DISPLAY=wayland-1
XDG_CONFIG_DIRS=/etc/xdg:/home/daniel/.local/share/flatpak/exports/etc/xdg:/var/lib/flatpak/exports/etc/xdg:/home/daniel/.nix-profile/etc/xdg:/etc/profiles/per-user/daniel/etc/xdg:/nix/var/nix/profiles/default/etc/xdg:/run/current-system/sw/etc/xdg
XDG_CURRENT_DESKTOP=sway
XDG_DATA_DIRS=/nix/store/jgwg4gn2fvpf4szs2flwcbgzjdcd3zsp-sway-1.7/share:/nix/store/ndg5l6magi63y734lvn1hsqr3awnmvjj-gnome-mimeapps/share:/nix/store/vdx1syjnzl7w45m7m93hb2x7imkiw6v7-desktops/share:/home/daniel/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/home/daniel/.nix-profile/share:/etc/profiles/per-user/daniel/share:/nix/var/nix/profiles/default/share:/run/current-system/sw/share:/nix/store/nflqcb2ffd52myw3gfl0xgqjxqxbrdz4-gnome-shell-42.1/share/gsettings-schemas/gnome-shell-42.1:/nix/store/llp402bj8nx0zsabkaygb3f9k4xfvrkb-gnome-shell-extensions-42.1/share/gsettings-schemas/gnome-shell-extensions-42.1
XDG_DESKTOP_PORTAL_DIR=/nix/store/rys85mif4wlny9rhsi7b5xzp306550a6-xdg-portals/share/xdg-desktop-portal/portals
XDG_RUNTIME_DIR=/run/user/1000
XDG_SEAT=seat0
XDG_SESSION_CLASS=user
XDG_SESSION_ID=2
XDG_SESSION_TYPE=wayland
XDG_VTNR=5
_SWAY_WRAPPER_ALREADY_EXECUTED=1
---
> tmux show-options -g update-environment
update-environment[0] DISPLAY
update-environment[1] KRB5CCNAME
update-environment[2] SSH_ASKPASS
update-environment[3] SSH_AUTH_SOCK
update-environment[4] SSH_AGENT_PID
update-environment[5] SSH_CONNECTION
update-environment[6] WINDOWID
update-environment[7] XAUTHORITY
update-environment[8] XDG_CURRENT_DESKTOP
update-environment[9] SWAYSOCK
update-environment[10] WAYLAND_DISPLAY
```

**Comment by nicm** [comment]
How did you *try* to make it go away?

**Comment by joshbode** [comment]
Thank you for the new release, @nicm :)

Just a hint for anyone else who uses [`base16-shell`](https://github.com/chriskempson/base16-shell/) to manage their themes, and are scratching their head like me as to why the theme no longer updates immediately (but does update after restarting your terminal program).

It appears to be [this section](https://github.com/chriskempson/base16-shell/blob/master/templates/default.mustache#L31=) in the shell scripts that apply the colours:
```shell
...
if [ -n "$_TMUX" ]; then
  # Tell tmux to pass the escape sequences through
  # (Source: http://permalink.gmane.org/gmane.comp.terminal-emulators.tmux.user/1324)
  put_template() { printf '\033Ptmux;\033\033]4;%d;rgb:%s\033\033\\\033\\' $@; }
  put_template_var() { printf '\033Ptmux;\033\033]%d;rgb:%s\033\033\\\033\\' $@; }
  put_template_custom() { printf '\033Ptmux;\033\033]%s%s\033\033\\\033\\' $@; }
elif [ "${TERM%%[-.]*}" = "screen" ]; then
  # GNU screen (screen, screen-256color, screen-256color-bce)
  put_template() { printf '\033P\033]4;%d;rgb:%s\007\033\\' $@; }
  put_template_var() { printf '\033P\033]%d;rgb:%s\007\033\\' $@; }
  put_template_custom() { printf '\033P\033]%s%s\007\033\\' $@; }
elif [ "${TERM%%-*}" = "linux" ]; then
  put_template() { [ $1 -lt 16 ] && printf "\e]P%x%s" $1 $(echo $2 | sed 's/\///g'); }
  put_template_var() { true; }
  put_template_custom() { true; }
else
  put_template() { printf '\033]4;%d;rgb:%s\033\\' $@; }
  put_template_var() { printf '\033]%d;rgb:%s\033\\' $@; }
  put_template_custom() { printf '\033]%s%s\033\\' $@; }
fi
...
```

I scanned the [CHANGES](https://raw.githubusercontent.com/tmux/tmux/3.3a/CHANGES) file but I couldn't see a specific item which fixes this, but it looks like the special treatment may no longer be necessary. 
@nicm - is this something you expected to change on this release? If so, I'll make a PR to `base16-shell` to disable this from version 3.3a and up :)

In any case, an easy workaround for me was to override the `TMUX` environment variable when applying themes to avoid the special treatment for `tmux`, e.g.:
```
$ TMUX="" base16_gruvbox-dark-medium
```

**Comment by nicm** [comment]
@joshbode support for OSC 4 was added in tmux 2.4 and the passthrough escape sequence was disabled by default in this release.

**Comment by joshbode** [comment]
thanks @nicm - I'll build that into the PR for `base16-shell` :)

**Comment by YellowOnion** [comment]
@nicm how would I try? Is there a command I'm unaware of? the documentation isn't exactly clean how to remove options from update-environment when it's not listed in update-environment.

**Comment by nicm** [comment]
`setenv -u`

**Comment by OilSubjectLoss7** [comment]
New update broke the click twice to copy text

Last version could copy 123.0.0.3 or nice.boat but the new version can only copy 123 or 0 3 or the dot(.) or nice or boat

Please fix

**Comment by nicm** [comment]
No, that is not possible and I do not see it happening.

**Comment by lfrancke** [comment]
Up until 3.2 I was using the solution from [this answer](https://unix.stackexchange.com/a/529049) to either attach or create a session.

This worked because in 3.2 `tmux attach -t ` (empty parameter) would result in this:
```
tmux attach -t
tmux: option requires an argument -- t
usage: attach-session [-dErx] [-c working-directory] [-f flags] [-t target-session]
```

In 3.3 this results in:
```
tmux attach -t
sessions should be nested with care, unset $TMUX to force
```

So it tries to attach to an existing session.
I looked at the changelog but couldn't find anything obvious in there that affects this but I do see that the man page has been changed (I believe) and it now says:

```If no client is specified, tmux attempts to work out the client currently in use; if that fails, an error is reported.```

Any idea/suggestion on how to achieve the old behavior in 3.3?
Thank you for your help!

**Comment by lfrancke** [comment]
Yes, I can solve it in bash if needed :)
I was wondering if there is a tmux way to "switch" to the old behavior of `-t` basically.

This is not a feature request, just a question. I skimmed the changelog and was just surprised by this new behavior.

**Comment by nicm** [comment]
I don't remember why it was changed or if it was intentional, would need to track down the change in Git. But it is probably best not to rely on this behaviour one way or the other, better to fully specify what you want.

**Comment by neural416** [comment]
I've added the cursor-style bar option to my config file but it doesn't seem to have any effect.  Am I doing something wrong?

Using xterm on linux.

**Comment by nicm** [comment]
What exactly did you put in .tmux.conf?

**Comment by neural416** [comment]
set-option -g cursor-style bar

I tried with other flags but I got: no current window when i started tmux

**Comment by nicm** [comment]
This looks like a bug, the cursor style is only being applied when the option is changed and only for existing panes, I'll fix it tomorrow.

**Comment by boris-petrov** [comment]
@topcat001 going back to a discussion from an [year ago](https://github.com/tmux/tmux/issues/2645#issuecomment-848368657) - I don't have a current match indicator in 3.3a. :( Did you manage to get that working?

**Comment by nicm** [comment]
@neural416 this should be fixed next time GitHub syncs if you build from master, or you can apply this change: [default_cursor.diff.txt](https://github.com/tmux/tmux/files/8925449/default_cursor.diff.txt)

**Comment by topcat001** [comment]
> @topcat001 going back to a discussion from an [year ago](https://github.com/tmux/tmux/issues/2645#issuecomment-848368657) - I don't have a current match indicator in 3.3a. :( Did you manage to get that working?

Hi @boris-petrov , we decided to drop this feature because it is tricky to implement correctly for all the edge cases, and neither vi or emacs has this. However, I kept the code in https://github.com/topcat001/tmux/commit/386b4ace014a8528359f146b8cff9925010fb644 if you're interested in what I was thinking.

**Comment by ffernand** [comment]
I've noticed that the behaviour of `search-backward-incremental` no longer allows you to jump from selection to selection (using `n` & `p` keys) after entering your search word on the prompt.  Reverting to v 3.2a restores this behaviour.

```
bind-key -n C-s copy-mode -e \; send -X clear-selection \; command-prompt -I "#{pane_search_string}" -ip "(search)" 'send -X search-backward-incremental "%%%"'
```
Not sure if it's a bug or if there's something missing in the above.

**Comment by topcat001** [comment]
I just tested this on a build off git and it seems to work for me? With `mode-keys` set to emacs, I can use `C-r` to start an incremental search, then press enter to commit. Next, `n` and `N` moves backwards and forwards, respectively, between the highlighted results.

**Comment by ffernand** [comment]
@topcat001 After your comment, I compiled tmux-git with a stripped down configuration and can confirm it works.  So I started excising my tmux.conf file to see where the problem might lie and I can reproduce with the following minimal setup...

```
# tmux.conf

# do not let EDITOR or VISUAL dictate status or mode keys in tmux.
setw -g status-keys emacs
setw -g mode-keys emacs

# the following keys in copy-mode will exit copy-mode
# and place the corresponding key in the terminal.
bind-key -T copy-mode n send -X cancel \; send-keys n
bind-key -T copy-mode p send -X cancel \; send-keys p

# when in the middle of a search, define keys to navigate through selections,
# otherwise cancel copy-mode and send corresponding key to the terminal.
bind-key -T copy-mode n if -F '#{search_present}' {send -X search-backward} {send -X cancel; send-keys n}
bind-key -T copy-mode p if -F '#{search_present}' {send -X search-forward} {send -X cancel; send-keys p}
```
When I'm in copy mode, I have a huge slew of characters (including n and p) whereby if they have no specific function then I exit out of copy-mode and begin typing on the command line right away, which is what lines like the following buy me...
```
bind-key -T copy-mode n send -X cancel \; send-keys n
bind-key -T copy-mode p send -X cancel \; send-keys p
```
However, if I'm in search-mode (if I can call it that), then the following lines should be able to adjust the behaviour of the characters n & p according to the `#{search_present}` lines... but it doesn't seem to work anymore in 3.3a.
```
bind-key -T copy-mode n if -F '#{search_present}' {send -X search-backward} {send -X cancel; send-keys n}
bind-key -T copy-mode p if -F '#{search_present}' {send -X search-forward} {send -X cancel; send-keys p}
```
I can comment out the first set of lines that just exits out of copy-mode and still reproduce the behavour since the second set just remaps the keys anyways.

I still can't tell yet what might be wrong in as far as syntax or behaviour in the last two lines.  I just know that it works as written in 3.2a

**Comment by topcat001** [comment]
@ffernand please change `search-backward` to `search-again`, etc., i.e., to commands which do not take an argument `<for>` and it will work I think.

**Comment by ffernand** [comment]
@topcat001 ... yes it does work!  Thank You!

I checked 3.2 manpage and can see that even in the previous version `search-backward` required an argument.  `search-again` and `search-reverse` work perfectly and I guess 3.3 tightens up the spec for both `search-backward` and `search-forward`.

Much appreciated for your help!

**Comment by boris-petrov** [comment]
> > @topcat001 going back to a discussion from an [year ago](https://github.com/tmux/tmux/issues/2645#issuecomment-848368657) - I don't have a current match indicator in 3.3a. :( Did you manage to get that working?
> 
> Hi @boris-petrov , we decided to drop this feature because it is tricky to implement correctly for all the edge cases, and neither vi nor emacs has this. However, I kept the code in [topcat001@386b4ac](https://github.com/topcat001/tmux/commit/386b4ace014a8528359f146b8cff9925010fb644) if you're interested in what I was thinking.

Oh, no... :disappointed: Can't it at least be returned to work in the "general" case and not work in all cases? I guess that would be good enough for most people... I really want to update my tmux from version 3.0 that I've been using for years now. :smile:

**Comment by krader1961** [comment]
@blissdev, passthrough sequences were disabled by default on 2022-02-15 by commit 5076beb00:

```
commit 5076beb009f761999a3b218a1a8d7cbfbc80ee03
Author: Nicholas Marriott <nicholas.marriott@gmail.com>
Date:   Tue Feb 15 13:11:29 2022 +0000

    Add an option (default off) to control the passthrough escape sequence.
    Like set-clipboard and allow-rename it is safer to forbid this by
    default.
```

See the recently added `allow-passthrough` option to enable the legacy behavior. However, I would question whether your `tig` command really requires, and should be using, the passthrough capability. 

P.S., I have no idea what `tig` command you are referring to. You really should include a link to the project that provides that command. I'm guessing that you are using https://jonas.github.io/tig/. Please, don't make us guess what tool you are using.

**Comment by blissdev** [comment]
Apologies @krader1961, that is the command I was referring to. And also I enabled the passthrough as I mentioned in my post, and I'm still having the issue.

**Comment by krader1961** [comment]
>  Apologies @krader1961, that is the command I was referring to. And also I enabled the passthrough as I mentioned in my post, and I'm still having the issue.

No, you did not mention enabling passthrough in this issue. You need to prove that you have enabled passthrough mode via a tmux server log that shows you did so. I think you are full of shit.

**Comment by nicm** [comment]
@blissdev this is more likely due to change in the default `TERM`, if you are on macOS perhaps you are hitting the Homebrew build problem described at the start of this issue, or if not then it seems likely there is a bug in `tig` that means it does not like `tmux-256color`.

**Comment by imomaliev** [comment]
After update, window-linked hook does not see environment variables set by set-environment on session-create. Is this a bug?

**Comment by topcat001** [comment]
> Oh, no... disappointed Can't it at least be returned to work in the "general" case and not work in all cases? I guess that would be good enough for most people... I really want to update my tmux from version 3.0 that I've been using for years now. smile

@boris-petrov No plans at the moment.

**Comment by broesler** [comment]
I see the change:

* Export TERM_PROGRAM and TERM_PROGRAM_VERSION like various other terminals.

I have a few scripts that rely on `$TERM_PROGRAM == iTerm.app`, that are now broken. Is there workaround to detect that I am running tmux from iTerm?

**Comment by nicm** [comment]
@brosler `tmux display -p '#{client_termtype}'` - but it will change if you attach from a different terminal.

**Comment by BrianGallew** [comment]
Not sure what's happening, but this new build seems to break the iTerm2 integration.  Trying to use `tmux -CC attach` results in a macOS notification that "A tmux pane terminated immediately after creation".
macOS 10.15.7
iTerm2: 3.4.15
Reverting to 3.2a (I honestly didn't try 3.3) makes it work again.

**Comment by nicm** [comment]
@BrianGallew is this https://github.com/tmux/tmux/issues/3199?

**Comment by BrianGallew** [comment]
Yes, it is, sorry, somehow my searching didn't find that one.

**Comment by vamshikk** [comment]
Hello,

I see that select-word has been fixed in tmux-3.3a to behave like it does in Vim ie it selects only the word. Is there a way to get the old behavior? 

In my work flow which I have used for years I have been using 'select-word' to copy things quickly from the terminal (paths, complex words with hypens, dots and underscores etc). This is a my most time saving shortcut. With the select-word behavior changed, I am not able to do that anymore. I am testing out some work arounds but they seem to break in certain scenarios. 

Any tips or help would be appreciated.

VK

**Comment by vamshikk** [comment]
@topcat001 Thank you! I was not aware of this option. I guess I didn't search hard enough! I love the fine grain control that it provides.

**Comment by imomaliev** [comment]
After update, `window-linked` hook does not see environment variables set by `set-environment` on `session-create`. Is this a bug?

**Comment by trallnag** [comment]
Offtopic: Where can I ask general questions about tmux? I have a weird "bug(?) with my Tmux configuration where the scrollback indicator gets duplicated on long lines.

![image](https://user-images.githubusercontent.com/24834206/179393748-8b9fabd1-ae79-47bd-830a-bbb758a1a50b.png)

**Comment by systemmonkey42** [comment]
What is the value of $TERM inside and outside of tmux?  Most display issues result from issues with terminal type settings.

**Comment by trallnag** [comment]
Outside, it's xterm-256color, inside it's screen-256color. Thanks for the hint. I will try playing around with the values. The problem disappears when I start tmux as root (so without all my custom config). There TERM is set to tmux-256color.

**Comment by trallnag** [comment]
@systemmonkey42, that was indeed the reason! Wonder why I set it to screen-256color. I think it had to do with getting sRGB to work properly. Thanks

**Comment by systemmonkey42** [comment]
@trallnag 
To answer your first question, there is a wiki section in the github repository which can help you find additional resources.
    https://github.com/tmux/tmux/wiki

**Comment by patrickpichler** [comment]
Is it on purpose that new tmux sessions do no longer inherit the environment variables from the 
parent process? 

I've tried searching in the changelog, but it is not mentioned anywhere.

**Comment by systemmonkey42** [comment]
@patrickpichler Is this something that has changed for you?  What is different to the previous behaviour?

**Comment by patrickpichler** [comment]
Yes, before the following was happening:
```
# this is the first tmux session
$ export TEST=1
$ tmux
$ echo $TEST
> 1

# in another terminal window
$ export TEST=2
$ tmux
$ echo $TEST
> 2
```

Now the following is happening
```
# this is the first tmux session
$ export TEST=1
$ tmux
$ echo $TEST
> 1

# in another terminal window
$ export HELLO=world
$ tmux
$ echo $HELLO
>
```
I hope you can follow what my problem is :sweat_smile:

**Comment by systemmonkey42** [comment]
I don't know if that was a change, but if it was you must be coming from a very old version.

Try adding the following to your TMUX conf file, and re-testing.
```
set-option -ga update-environment TEST
set-option -ga update-environment HELLO
```

TMUX Is very picky about what environment variables get slurped up, which is good for security and avoiding environment pollution.

**Comment by patrickpichler** [comment]
That is interesting! I have no idea how it worked before the way it did :sweat_smile: 
The one thing I personally find a bit strange is, that it inherits all environment variables
when the first session is started. 

But thank you for the information :blush:

**Comment by asmodeus812** [comment]
Any idea why the `list-panes -f` prints out that -f is unknown option ? It is part of the docs, but the usage shows that it is not ????
![image](https://user-images.githubusercontent.com/15955811/187347985-d793a708-9a2e-4ae6-8a86-37d88a9419b2.png)
![image](https://user-images.githubusercontent.com/15955811/187347967-ab05a179-1f7e-43e4-88fd-e59421228ef9.png)

**Comment by krader1961** [comment]
@asmodeus812: Works for me. The `-f` option was added 2020-04-12. So I can only conclude you are using an old version of tmux. What does `tmux -V` output? And why do you think your problem is relevant to this issue?

**Comment by asmodeus812** [comment]
Oh, i see mine is 3.0, i had a feeling that might be the case. But that is the latest available package in apt for ubuntu 20. Tough luck i guess, i found a workaround to get the information from list-panes with a filter. It would be great if there is any mention in the manual which version supports which flags.

**Comment by revolter** [comment]
Can the `tmux-256color.src` file that's created by the command `/*/*/Cellar/ncurses/6.3/bin/infocmp -x tmux-256color >tmux-256color.src` be safely deleted afterwards?

**Comment by nicm** [comment]
Yes once you have installed it with `tic` you can remove it.

**Comment by systemmonkey42** [comment]
Where can I find the source for `tmux-256color` and other tmux terminal types in infocmp or termcap format?  Are they part of tmux (I can't find them in the repo) or are they 3rd party?

If they aren't part of the tmux source repo, why not?

**Comment by nicm** [comment]
terminfo is part of ncurses (in misc/terminfo.src).

**Comment by jpe90** [comment]
Is there a way to capture the output from the `shell-command` passed to `display-popup`? My use case is that I'm trying to find a way to do something like `tmux display-popup -E -h 75% fzf`, and after I select the file and the popup closes I'd want the result printed as if I had invoked `fzf` normally.

**Comment by nicm** [comment]
No you will need to use a temporary file or have the command call `tmux setenv`/`tmux set` or similar.

**Comment by systemmonkey42** [comment]
@jpe90 fzf comes with a helper script called fzf-tmux.  Read that as a guide on how to do what you want.  Or better yet, use it, since that is what it's for.

**Comment by jpe90** [comment]
@nicm Thank you

@systemmonkey42 I had seen that before but I didn't notice the popup option inside there, thank you!

**Comment by adoyle-h** [comment]
I tried `cursor-colour` and `cursor-style` in .tmux.conf. But both not work. No error thrown.

<img width="228" alt="CleanShot 2022-09-21 at 15 32 13@2x" src="https://user-images.githubusercontent.com/1998490/191442724-7fad7a2b-5902-4a73-b8d7-5e32dd5b9eee.png">


```
set -g cursor-colour green
set -g cursor-style blinking-bar
```

I set `cursor-colour` and `cursor-style` when tmux server running. It works.

<img width="285" alt="CleanShot 2022-09-21 at 15 32 31@2x" src="https://user-images.githubusercontent.com/1998490/191442780-e968374e-d252-4b49-9d17-7a4208ed0fa4.png">

<img width="201" alt="CleanShot 2022-09-21 at 15 32 44@2x" src="https://user-images.githubusercontent.com/1998490/191442828-af6709b4-db51-4f3d-987a-96be4ec3d8d6.png">

tmux -V: tmux 3.3a

**Comment by huyz** [comment]
I'm confused about the state https://github.com/tmux/tmux/pull/3274 . @nicm's comments made it sound like it was accepted but it looks to me like the PR was closed without merging. Am I misunderstanding?

**Comment by nicm** [comment]
GitHub PRs are never merged, see https://github.com/tmux/tmux/wiki/Contributing#source-code-and-development-process.

**Comment by imomaliev** [comment]
_Off-topic_: This was previously brought up https://github.com/tmux/tmux/issues/3239#issuecomment-1167160625, but there was no response. If it is possible I think it would be great to add `Co-authored-by` to git commits created by script from OpenBSD CVS https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors to acknowledge contributions

**Comment by nicm** [comment]
Everyone with an identifiable name is credited in the commit message already and I don't want the extra work and clutter of filling commit messages with whatever other bits of GitHub decoration is in fashion this year.

If instead you'd like to build and maintain a `CONTRIBUTORS` file then that could work.

**Comment by asmodeus812** [comment]
Hello, i have an issue with 3.3a which is not observed in 3.0a for example - i am running in alacritty and have defined the following sequence 

```yaml
  - { key: Back,      mods: Control,    chars: "\x1b[08;5u" }
  - { key: Delete,    mods: Control,    chars: "\x1b[127;5u" }
```
In the older version when i run showkey -a and press ctrl + backspace or delete i see exactly those key sequences, in 3.3 though for ctrl + backspace i see ^? and for backspace i also see ^?. What might be the issue here ? For the ctrl + del i don't even see anything getting printed in the console when in tmux, again in 3.0 it works, i get the defined sequence. Note that outside of tmux it all works as expected just from alacritty and bash.

**Comment by arcnmx** [comment]
Since updating, `set -g mouse off` doesn't seem to actually disable anything. Mouse events can control programs inside panes regardless (vim for example allows me to select around, scroll, etc). Is there a new way to disable mouse control?

I'm using tmux 3.3a with rxvt-unicode (urxvt) v9.30 in xorg.

**Comment by tobiabocchi** [comment]
Hello, I am not sure if this is the right place to ask, so sorry in advance if this does not belong here.
I am trying to achieve a behaviour similar to the one of `prefix :list-keys` but through a shell script, I have some text that I want to display in the current pane in a "disposable" pane in copy-mode

**Comment by topcat001** [comment]
> Hello, I am not sure if this is the right place to ask, so sorry in advance if this does not belong here. I am trying to achieve a behaviour similar to the one of `prefix :list-keys` but through a shell script, I have some text that I want to display in the current pane in a "disposable" pane in copy-mode

Hi @tobiabocchi this can be achieved using:
```
tmux run-shell -b <command>
```

**Comment by ltWolfik** [comment]
Hello! I have session A and session B (attached). If bell (`tput bel` or `echo -e "\a"` for example) is produced in session B - the bell is passed through to the terminal (I've setup `set -g visual-bell both`). But if bell is produced in hidden session A - the bell is not passed (I've setup the terminal to produce sound/system notification on a bell). There is a way to pass bell through from hidden /detached session?
Thanks in advance.

OpenSUSE Tumbleweed
KDE Plasma 5.25.5
Konsole v22.08.1
tmux 3.3a

**Comment by nicm** [comment]
No, you can only be alerted for bells in the attached session, not others.

**Comment by huyvohcmc** [comment]
When I use nvim alone, the clipboard nvim detects is pbcopy (I'm on macos), but when I use tmux, the clipboard nvim detects is `tmuxclipboard`, and I can just paste from system clipboard to nvim in normal mode any more. Can I disable `tmuxclipboard`?

**Comment by tobiabocchi** [comment]
> > Hello, I am not sure if this is the right place to ask, so sorry in advance if this does not belong here. I am trying to achieve a behaviour similar to the one of `prefix :list-keys` but through a shell script, I have some text that I want to display in the current pane in a "disposable" pane in copy-mode
> 
> Hi @tobiabocchi this can be achieved using:
> 
> ```
> tmux run-shell -b <command>
> ```

Hey @topcat001, thank you for helping me out! Your suggestion works for something like `tmux run-shell "cat /etc/hosts"`. 
I'd like to have a binding that runs a script in a split, the script retrieves some text based on user input, then it displays it inside the pane in view mode..
So far what I made works except that I had to put a `read` statement at the end of the script so that it does not suddenly exits view mode.
If my explanation was not clear the script is [here](https://pastebin.com/JKi9QNje)

**Comment by thiagowfx** [comment]
It would be nice if there was an elegant way to make `set -g allow-passthrough on` fail silently in older versions of `tmux`. I set it to `on` in my dotfiles, however whenever I `ssh` into Debian (which takes a long time to upgrade packages, by design) it fails:

```
/home/pi/.tmux.conf:72: invalid option: allow-passthrough
```

There are many Stack Overflow answers that suggest to parse the output of `tmux -V` and pipe it into `bc` or similar + glue into `if-shell` in order to check if the tmux version is greater than X.Y to set the option silently, but they're all a bit of a hack. It would be nice of vanilla `tmux` could support unrecognized options for older versions of `tmux`.

For example, OpenSSH has this [`IgnoreUnknown`](https://www.unixtutorial.org/ssh-bad-configuration-option-usekeychain/) setting that serves exactly this purpose.

**Comment by nicm** [comment]
https://github.com/tmux/tmux/wiki/FAQ#how-can-i-make-tmuxconf-portable-between-tmux-versions

**Comment by thiagowfx** [comment]
Thanks! I’ll look into updating some of these old answers to mention that.

**Comment by itsjfx** [comment]
is it possible to suppress specific error messages, or maybe them in general?
e.g. `Can't find window: x` when attempting to select an invalid window?

<https://github.com/tmux/tmux/blob/36896f6dd0d49d91bcc9c398c68a33818626532c/cmd-find.c#L1232>

I can see a check on the `CMD_FIND_QUIET` flag but not sure how to activate this?

**Comment by topcat001** [comment]
I don't think this is exposed to the user. Some of the commands (e.g., `show-options`) use it internally to run a command in `CMD_FIND_CANFAIL` mode.

**Comment by genki** [comment]
Is there a way to make a pane "display only"? 
It means both read only and being ignored from pane navigation.
I supposed that to display indication such as `top`, `npm run dev`.

**Comment by systemmonkey42** [comment]
@genki, I use a combination of things to achieve this.

Firstly, a hotkey to toggle the "display only" mode. I use Prefix-<kbd>`</kbd> (backtick)
```
# Toggle pane input (blocks and unblocks keyboard input to pane)
bind-key -T prefix       `                             if-shell -F "#{pane_input_off}" "select-pane -e" "select-pane -d"
```

Secondly, I define a command alias to handle the equivalent of "next-pane, but skip disabled panes" and "previous-pane, skipping disabled panes"
```
set-option -s command-alias[100] "next-active-pane=if-shell -F '#{P:#{?#{pane_input_off},,1}}' { select-pane -t '{next}'; if-shell -F \"#{pane_input_off}\" 'next-active-pane' }"
set-option -s command-alias[101] "previous-active-pane=if-shell -F '#{P:#{?#{pane_input_off},,1}}' { select-pane -t '{previous}'; if-shell -F \"#{pane_input_off}\" 'previous-active-pane' }"
```
The use of a command-alias here is essentially magic, because of the way it handles argument expansion and iteration (techically tail-end recursion) which isn't possible in a binding.

Finally, I use <kbd>Ctrl-TAB</kbd> and <kbd>Ctrl-Shift-Tab</kbd> to navigate forwards and backwards among panes.
```
# Ctrl-Tab (Xterm)
bind-key -T root         C-Tab                           next-active-pane

# Ctrl-Shift-Tab (XTerm)
bind-key -T root         C-S-Tab                         previous-active-pane
```

If you use the default <kbd>Prefix-[up/down/left/right]</kbd> keys, it will still go to disabled panes, which makes it easy to re-enable them. (You can also simply click on them to activate them before enabling them if you have mouse support enabled.)

While this works perfectly for me, a number of TMUX bugs needed to be resolved (including #2842, #2868 and #2554) before it was stable.   As a result, you many find this doesn't work for you if you arent using a fairly recent build.

**Comment by genki** [comment]
@systemmonkey42 Thank you for detailed explanation.
I will try to mimic it :)
I wish it would be a built-in feature.

**Comment by systemmonkey42** [comment]
> I wish it would be a built-in feature.

An option to `select-pane` which skips disabled panes would help....

       select-pane -t {next-active}

or allow existing next/prev/first/last/U/D/L/R options to skip disabled panes

        select-pane -t {next} -A

or for more precise selection,  a filter to limit which panes should be considered when navigating...

      select-pane -t {next} -f '#{?#{pane_input_off},0,1}'

as an example...

**Comment by gordonbazeley** [comment]
Hi there - I've recently updated to this version. I connect via tmuxinator and since this version I'm getting a "can't find session: [session name]" error when running "mux [session name]" - any ideas please?

**Comment by vlaw** [comment]
# [ASKING for help]   Wield problem , Totally no clue

After upgrade TMUX, `curl/wget` is not working at all, but work normal if exit tmux env.

ERROR log is this bellow:

```shell
$ curl -vk -I https://www.github.com
*   Trying 20.205.243.166:443...
* connect to 20.205.243.166 port 443 failed: Software caused connection abort
* Failed to connect to www.github.com port 443 after 214 ms: Couldn't connect to server
* Closing connection 0
curl: (7) Failed to connect to www.github.com port 443 after 214 ms: Couldn't connect to server
```

I figure out there is something about openssl, but can't find how to dig deeper.
and the command bellow works fine  outside tmux.

```shell
$ echo 'quit' | openssl s_client -connect github.com:443 -msg
CONNECTED(00000005)
>>> TLS 1.3 Handshake [length 011a], ClientHello
    01 00 01 16 03 03 86 3a 39 3e 2f a8 79 bb b6 a6
    92 e8 16 3e 8d af a2 69 db d6 17 e3 97 88 51 89
    43 63 46 6f c7 8e 20 ae d0 20 46 20 1f 39 d6 29
    d3 4a 05 14 1a ff 44 23 6c 44 95 2f 09 fa 0f d2
    86 26 2e 14 b1 3b 86 00 62 13 02 13 03 13 01 c0
    30 c0 2c c0 28 c0 24 c0 14 c0 0a 00 9f 00 6b 00
    39 cc a9 cc a8 cc aa ff 85 00 c4 00 88 00 81 00
    9d 00 3d 00 35 00 c0 00 84 c0 2f c0 2b c0 27 c0
    23 c0 13 c0 09 00 9e 00 67 00 33 00 be 00 45 00
    9c 00 3c 00 2f 00 ba 00 41 c0 11 c0 07 00 05 00
    04 c0 12 c0 08 00 16 00 0a 00 ff 01 00 00 6b 00
    2b 00 09 08 03 04 03 03 03 02 03 01 00 33 00 26
    00 24 00 1d 00 20 3c 82 ce 0e bb d3 20 f0 df f2
    ab 36 bd b2 99 52 a6 39 e1 cf fd 6d 85 52 46 2d
    55 95 5f 35 79 13 00 0b 00 02 01 00 00 0a 00 0a
    00 08 00 1d 00 17 00 18 00 19 00 23 00 00 00 0d
    00 18 00 16 08 06 06 01 06 03 08 05 05 01 05 03
    08 04 04 01 04 03 02 01 02 03
write:errno=53
---
no peer certificate available
---
No client certificate CA names sent
---
SSL handshake has read 0 bytes and written 287 bytes
---
New, (NONE), Cipher is (NONE)
Secure Renegotiation IS NOT supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
    Protocol  : TLSv1.3
    Cipher    : 0000
    Session-ID:
    Session-ID-ctx:
    Master-Key:
    Start Time: 1676548268
    Timeout   : 7200 (sec)
    Verify return code: 0 (ok)
---
```

I have tried to remove all `.tmux.conf`, still not working. It's killing me

**Comment by topcat001** [comment]
Just to cover some obvious things, did you test this on a freshly started tmux server? I could not reproduce the issue on my usual usage scenarios (Arch Linux, OpenBSD, and macOS).

**Comment by joshbode** [comment]
@vlaw are there any key environment variables that are/aren't being established in the `tmux` session?

```bash
$ printenv | grep -Pi "^(PROXY|OPENSSL)[^=]*?=" | sort
```

Also, which version of `curl` and `openssl`, e.g.? Check this inside and outside the `tmux` session
```
$ curl --version
curl 7.87.0 (x86_64-pc-linux-gnu) libcurl/7.87.0 OpenSSL/3.0.8 zlib/1.2.13 brotli/1.0.9 zstd/1.5.2 libidn2/2.3.2 libssh2/1.10.0 nghttp2/1.51.0
Release-Date: 2022-12-21
Protocols: dict file ftp ftps gopher gophers http https imap imaps mqtt pop3 pop3s rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS brotli GSS-API HSTS HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile libz NTLM NTLM_WB SPNEGO SSL threadsafe TLS-SRP UnixSockets zstd

$ openssl version
OpenSSL 3.0.8 7 Feb 2023 (Library: OpenSSL 3.0.8 7 Feb 2023)
```

**Comment by vlaw** [comment]
> ```shell
> printenv | grep -Pi "^(PROXY|OPENSSL)[^=]*?=" | sort
> ```

Got nothing~.

> Also, which version of `curl` and `openssl`, e.g.? Check this inside and outside the `tmux` session

I `diff`ed the two console output, and the outputs are pasted bellow.
There is no difference inside and outside the `tmux` session.

```shell
$ curl --version
curl 7.87.0 (x86_64-apple-darwin22.1.0) libcurl/7.87.0 (SecureTransport) OpenSSL/1.1.1t zlib/1.2.11 brotli/1.0.9 zstd/1.5.4 libidn2/2.3.4 libssh2/1.10.0 nghttp2/1.51.0 librtmp/2.3
Release-Date: 2022-12-21
Protocols: dict file ftp ftps gopher gophers http https imap imaps ldap ldaps mqtt pop3 pop3s rtmp rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS brotli GSS-API HSTS HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile libz MultiSSL NTLM NTLM_WB SPNEGO SSL threadsafe TLS-SRP UnixSockets zstd
$ openssl version -a
LibreSSL 3.3.6
built on: date not available
platform: information not available
options:  bn(64,64) rc4(16x,int) des(idx,cisc,16,int) blowfish(idx)
compiler: information not available
OPENSSLDIR: "/private/etc/ssl"
```

**Comment by vlaw** [comment]
> Just to cover some obvious things, did you test this on a freshly started tmux server

Yes. 
I tried removing config files of tmux(`tmux.conf`, `tmux.conf/`,
and config files of zsh like `.zshenv`, `.zshrc` etc, the result is the same.

I even created a new User, and got the same error in that fresh new env.

**Comment by krader1961** [comment]
@vlaw, It is almost a certainty that your problem is due to different environment variables in the two scenarios. There is literally nothing that Tmux can do to cause your problem other than not provide the expected env vars. @joshbode should not have provided such a specific command to run since they presumed to know which env vars are likely to be relevant.

They asked if you tested "this on a freshly started tmux server.". You replied that your removed a bunch of configuration files but did not tell us that you killed the existing Tmux server (by running `tmux kill-server`) before performing another test.

I suggest simply running the `env` command in the Tmux and non-Tmux environments and comparing their output. I am confident that will yield a crucial difference.

**Comment by vlaw** [comment]
> @vlaw, It is almost a certainty that your problem is due to different environment variables in the two scenarios. There is literally nothing that Tmux can do to cause your problem other than not provide the expected env vars. @joshbode should not have provided such a specific command to run since they presumed to know which env vars are likely to be relevant.
> 
> They asked if you tested "this on a freshly started tmux server.". You replied that your removed a bunch of configuration files but did not tell us that you killed the existing Tmux server (by running `tmux kill-server`) before performing another test.
> 
> I suggest simply running the `env` command in the Tmux and non-Tmux environments and comparing their output. I am confident that will yield a crucial difference.

I do run `tmux kill-server`, and `pkill -9 tmux`.

This is the env diff:

```shell
$ diff ~/tmp/env.txt ~/tmp/env.tmux.txt > ~/tmp/env.diff

83c83
< SHLVL=1
---
> SHLVL=2
85,88c85,88
< TERM=xterm-256color
< TERM_PROGRAM=iTerm.app
< TERM_PROGRAM_VERSION=3.4.19
< TERM_SESSION_ID=w0t0p1:0935D52A-A4A3-4821-A8DA-F82A2E7A41D6
---
> TERM=screen-256color
> TERM_PROGRAM=tmux
> TERM_PROGRAM_VERSION=next-3.4
> TERM_SESSION_ID=w0t0p1:6A954574-F7C7-4863-AB34-921BCE1D64DC
89a90,92
> TMUX=/private/tmp/tmux-501/default,28042,1
> TMUX_PANE=%1
> TMUX_PLUGIN_MANAGER_PATH=/Users/******/.tmux/plugins/
```

**Comment by krader1961** [comment]
@vlaw, I don't see any reason why the `SHLVL` value should be different in those two scenarios. Which makes me question the rest of the env var differences you reported. There is literally nothing, other than env var differences, why Tmux would cause your `curl` command failure.

You need to a) kill the existing Tmux server, b) start a new one with `tmux -f /dev/null`, c) reproduce your `curl` example. Even then it is likely your problem is due to a difference in the env vars. You might, for example, have some shell startup scripts which test whether they are being run inside a Tmux session. That is extremely unlikely but illustrates the type of thing that can cause the behavior your are seeing.

**Comment by joshbode** [comment]
> @joshbode should not have provided such a specific command to run since they presumed to know which env vars are likely to be relevant.
True, but I did read the man pages for `curl` and `openssl` first, so it wasn't totally uninformed on which environment variables should have an effect.
Also, I've seen people post some pretty sensitive info inadvertently by including everything `printenv` outputs :)

> I don't see any reason why the `SHLVL` value should be different in those two scenarios
That's how it is on my machine:
```
# outside tmux session
$ echo $SHLVL
1

# inside tmux session
$ echo $SHLVL
2
```

**Comment by vlaw** [comment]
> You need to a) kill the existing Tmux server, b) start a new one with `tmux -f /dev/null`, c) reproduce your `curl` example. Even then it is likely your problem is due to a difference in the env vars. You might, for example, have some shell startup scripts which test whether they are being run inside a Tmux session. That is extremely unlikely but illustrates the type of thing that can cause the behavior your are seeing.

I just create a new account, still the same result.
Running `curl` outside tmux session works fine, but inside tmux  get error as before.

The steps are:

1. reboot
2. switch to new account just created
3. run `tmux kill-server` just in case
4. run `ps -ef | grep -i tmux | grep -v grep` , got nothing
5. run `env | sort` inside ( using `tmux -f /dev/null`) and outside tmux session ( works fine )

```diff
8c8
< SHLVL=1
---
> SHLVL=2
10,12c10,12
< TERM=xterm-256color
< TERM_PROGRAM=Apple_Terminal
< TERM_PROGRAM_VERSION=447
---
> TERM=screen-256color
> TERM_PROGRAM=tmux
> TERM_PROGRAM_VERSION=next-3.4
14a15,16
> TMUX=/private/tmp/tmux-502/default,4393,0
> TMUX_PANE=%0
```

```shell
# inside tmux
testabc@VLAW-MBP ~ % curl -vk -I https://www.github.com
*   Trying 20.205.243.166:443...
* Connected to www.github.com (20.205.243.166) port 443 (#0)
* ALPN: offers h2
* ALPN: offers http/1.1
* (304) (OUT), TLS handshake, Client hello (1):
* error:02FFF035:system library:func(4095):Software caused connection abor
* Closing connection 0
curl: (35) error:02FFF035:system library:func(4095):Software caused connection abor
```

And one more thing, I switch to using `screen`, works fine also, WTF

**Comment by joshbode** [comment]
I've got a thought... could it be that macOS is limiting the capabilities of `tmux`, e.g. network access (or perhaps firewall)? I've seen that with GUI applications, but possibly for regular executables too...

**Comment by vlaw** [comment]
> I've got a thought... could it be that macOS is limiting the capabilities of `tmux` (and its child processes), e.g. network access (or perhaps firewall)? I've seen that with GUI applications where you need to authorise them to use the network, but possibly for regular executables too...

It's TRUE! Looks like  "all network is blocked " indeed.
I'm trying to figure out which security setting does such a evil thing.
The firewall is about  "allows users to block incoming connections on their Macs", but in my case it's about outcoming connection.

And it's also very wield that the `screen` work fine while I do not change any security settings.

```shell
# inside tmux
$ echo -e "GET / HTTP/1.0\r\nHost: www.github.com\r\n" | netcat www.github.com 80
write(net): Software caused connection abort

# outside tmux
$ echo -e "GET / HTTP/1.0\r\nHost: www.apple.com\r\n" | netcat www.apple.com 80
HTTP/1.0 301 Moved Permanently
Server: GHost
Content-Length: 0
Location: https://www.apple.com/
Cache-Control: max-age=0
Expires: Fri, 17 Feb 2023 09:43:15 GMT
Date: Fri, 17 Feb 2023 09:43:15 GMT
X-Cache: TCP_MISS from a210-192-119-20.deploy.akamaitechnologies.com (AkamaiGHost/11.0.0-46340752) (-)
Connection: close
strict-transport-security: max-age=31536000
Set-Cookie: geo=CN; path=/; domain=.apple.com
```

**Comment by vlaw** [comment]
> @vlaw I'm guessing it's something like `screen` is `/usr/bin/screen` and the binary is signed as part of the macOS base install? Where is `tmux` and how was it installed?

```shell
$ tmux -V
tmux 3.3a

$ which -a screen
/usr/local/bin/screen
/usr/bin/screen
$ /usr/local/bin/screen --version
Screen version 4.09.00 (GNU) 30-Jan-22
$ /usr/bin/screen --version
Screen version 4.00.03 (FAU) 23-Oct-06
```

The `echo ***` command works fine in the two versions of `screen` :(
I installed this homebrew version of `screen` in purpose， and found out works fine in both.

The `tmux` version is `tmux 3.3a`, which is install by homebrew.

**Comment by topcat001** [comment]
Check the privacy tab under system settings on macOS. It usually lists every app which has requested access to files or network and allows restoring access.

**Comment by vlaw** [comment]
> Check the privacy tab under system settings on macOS. It usually lists every app which has requested access to files or network and allows restoring access.

Didn't find anything related. I added tmux in the “Full Disk Access” and “Files and  Folders” etc, still got no  lucky.

**Comment by vlaw** [comment]
> > Check the privacy tab under system settings on macOS. It usually lists every app which has requested access to files or network and allows restoring access.
> 
> Didn't find anything related. I added tmux in the “Full Disk Access” and “Files and Folders” etc, still got no lucky.

It turned out to be some faults of some IT guys in our company, :(
Sorry for bothering

**Comment by ConnorWay32** [comment]
Is there a tmux variable for the hostname of a remote server you're SSHed into? If I have a tmux session, and ssh to a server from inside it, I'm wondering if there's a way to use the hostname of that server in a formatting section. #{host} displays the hostname of the client, and I'd imagine that any environment variables would be based on the client machine as well. Am I missing something, or does anyone know of a workaround?

**Comment by systemmonkey42** [comment]
> Is there a tmux variable for the hostname of a remote server you're SSHed into?

tmux does not monitor what you type into the shell.  Your best option is to write a wrapper script for `ssh`.

The wrapper script could then use tmux to set a "user option" in the active pane containing the host name, and clear the option when ssh is done.

```
tmux set-option -p @remote_host "$1"
```

Your custom status format could then use the pane-specific user variable to manipulate the status.

**Comment by krader1961** [comment]
> tmux does not monitor what you type into the shell. Your best option is to write a wrapper script for ssh.

I was just about to write the same thing. I'll also point out that even if it did that wouldn't help since there is absolutely no requirement that the SSH command be named `ssh`. For that matter, I might have a wrapper script around the `ssh` command that has the hostname(s) embedded within it so that the hostname doesn't even appear on the command line. I'm always curious about the person's mental model of how things like `ssh` work when I see questions like the one we're replying to.

**Comment by ConnorWay32** [comment]
> Save a script to /usr/local/bin/onconnect for example
> 
> ```
> #!/bin/bash
> 
> finished() {
>         [[ -n "${TMUX}" ]] && tmux set-option -pu @remote
> }
> 
> trap "finished" 0
> [[ -n "${TMUX}" ]] && tmux set-option -p @remote "$1"
> nc "$1" "$2"
> ```
> 
> Add the following to `~/.ssh/config`
> 
> ```
> Host *
>     ProxyCommand                 /usr/local/bin/onconnect %h %p
> ```

Thanks, that worked great.

> I'm always curious about the person's mental model of how things like `ssh` work when I see questions like the one we're replying to.

I'm fairly new to tmux and SSH. I assumed that since tmux could prevent SSH connections from being disrupted, that there was some deeper integration between tmux and SSH, or at least tmux and the remote machine.

**Comment by systemmonkey42** [comment]
> I assumed that since tmux could prevent SSH connections from being disrupted,

I'd love to know how you came to believe this.
TMUX has almost* zero knowledge of ssh, and does nothing special with SSH compared to any other program.

 <sub>* (tmux imports certain environment variables when you connect.  A couple of these are SSH specific, others are X11 or KERBEROS specific.)</sub>

**Comment by ConnorWay32** [comment]
I misunderstood how tmux prevented connection drop-offs, I just made the
assumption that tmux itself reconnected. Again, I don't have a thorough
understanding of how SSH, and tmux work, my experience so far has been
running tmux locally so I've never messed around with this stuff before.

**Comment by taupezen** [comment]
Hi,

Could someone provide an example of setting _main-pane-width_ option in _percentage_ ?
I was not able to set it, although in doc it is stated that : `If suffixed by ‘%’, this is a percentage of the window size`

**Comment by niksingh710** [comment]
is sixel support is being considered?

**Comment by topcat001** [comment]
@niksingh710 I believe the most up-to-date branch is my fork on github (https://github.com/topcat001/tmux/tree/sixel). I haven't had time to finish it up, but feel free to give it a try (you need to enable `allow-passtrough` in the config). I plan to get back on it soon so that some basic support can be merged. Please ignore my ugly hacks in the code if you do dig into it.

**Comment by vlappa** [comment]
I would like to synchronize-panes for a specific window only, is that possible? If not, should I issue a feature request?

**Comment by fjbaldon** [comment]
I need help with tmux crashing when selecting/pasting text. Where do I get the logs?

**Comment by nifisher** [comment]
After updating Suse Tubleweed a week or so ago to the latest packages (TMUX has been 3.3a for a while I assume) I have been getting frequent random crashes that bring down my tmux sessions.   I need to `reset` my console session to get rendering working again and then launch a new tmux session.

Dmesg has simply these two lines:
```
/home/user:: dmesg | egrep "tmux|Code"
[ 4499.516430] tmux: server[1647]: segfault at 0 ip 0000561e15b1b852 sp 00007ffcf2e22670 error 4 in tmux[561e15a89000+9d000] likely on CPU 3 (core 0, socket 6)
[ 4499.516463] Code: c0 74 3d 83 f8 01 75 5c 48 8b b9 08 0c 00 00 6a 00 45 31 c9 45 31 c0 31 c9 6a 00 31 c0 6a 00 6a 00 e8 c2 e2 f6 ff 48 83 c4 20 <80> 38 00 48 89 c3 75 16 48 83 c4 08 5b 5d c3 0f 1f 80 00 00 00 00

```
I'm assuming that this is an issue with my system and not tmux, but i'm not sure how to interpret this error message or where to start digging for more information (server.c doesn't have 1600 lines, so that doesn't seem to be a line number and/or file name).   Might have anyone have suggestions/guidance that they could provide?

**Comment by topcat001** [comment]
@nifisher Can you try git master? In the output, "tmux: server" is the name of the process, and [1647] is the pid. It has no connection to server.c.

**Comment by nifisher** [comment]
The pid makes sense... not sure why I didn't think of that.

After some swearing and installing of packages that I didn't know were not installed, `tmux next -3.4` is intalled.  Will let you know if i see the issue again.

**Comment by nifisher** [comment]
> `tmux next -3.4` is installed. Will let you know if i see the issue again.

3 days in and all seems well.   Thanks.

**Comment by npub1zenn0** [comment]
> There is now a menu when button 3 is pressed outside a popup border with a few items including the ability to convert a popup into a pane.

Is there a keybind I can use/create for converting the popup in a pane?

**Comment by systemmonkey42** [comment]
>Is there a keybind I can use/create for converting the popup in a pane?
Short answer is no.  Popups are very magic.

The code to display and handle the menu while a popup is visible is baked in, and the function to convert popups to panes is not exposed as a user function.

If it was, you would probably have to call it from a script inside the popup, since bindings are disabled while a popup is visible, and popups are attached to the CLIENT and not the "session" or "window" per se.

(Try opening a second client and connecting to the same session, you will be able to interact with the background window, but not the popup)

**Comment by jstkdng** [comment]
Hello, I'm developing an application that interfaces with tmux through running tmux commands, like `tmux list-clients ...`. I want to make my application faster so I was thinking to start communicating directly to tmux's socket. Is the socket communication documented anywhere?

**Comment by nicm** [comment]
https://github.com/tmux/tmux/wiki/Control-Mode

**Comment by jstkdng** [comment]
With that I'd need to start another process and pipe stdin and stuff. Is there no other way with using send/recv syscalls?

**Comment by awaken233** [comment]
```
set-option -g default-path "$PWD"
```
this option not work 🤕.

```
❯ tmux source-file ~/.tmux.conf
invalid option: default-path
```

**Comment by joshbode** [comment]
> ```
> set-option -g default-path "$PWD"
> ```
> 
> this option not work 🤕.
> 
> ```
> ❯ tmux source-file ~/.tmux.conf
> invalid option: default-path
> ```

It looks like this option was removed quite a while back:
  https://github.com/tmux/tmux/blob/4c60afde782cd024da5663d98c628c62efb013e0/CHANGES#L1571

**Comment by Docslims** [comment]
/*/*/Cellar/ncurses/6.3/bin/infocmp -x tmux-256color >tmux-256color.src

$ sudo /usr/bin/tic -x tmux-256color.src

**Comment by bubanoid** [comment]
Cannot install `tmux 3.3a` from release tarball following the instruction:

    ./configure && make
    sudo make install
    
no such file or directory: `./configure`. How can it be fixed?

**Comment by pataquets** [comment]
Browsed around docs trying to figure out the release schedule with no luck.
Is there any plan for cutting a new release?

**Comment by laixintao** [comment]
Hi does anyone got issues with mouse?

in tmux 3.2a, if I double click on an IP, it will mark the whole IP like this:

<img width="296" alt="image" src="https://github.com/tmux/tmux/assets/9675939/54f95814-874c-4cf4-917c-202059bd0a7a">

But in tmux 3.3a, if I do the same thing, the tmux will only mark one number:

<img width="209" alt="image" src="https://github.com/tmux/tmux/assets/9675939/75736b47-729d-4c17-9b21-9be9295118bc">

**Comment by tao12345666333** [comment]
@laixintao I'm using version 3.3a of tmux on Fedora, but I haven't encountered the same issue.

![Screenshot from 2023-11-24 20-45-08](https://github.com/tmux/tmux/assets/3264292/cfa1744e-3a0d-4114-9ecb-f9d5c2f92671)

**Comment by laixintao** [comment]
> @laixintao I'm using version 3.3a of tmux on Fedora, but I haven't encountered the same issue.
> 
> ![Screenshot from 2023-11-24 20-45-08](https://private-user-images.githubusercontent.com/3264292/285459712-cfa1744e-3a0d-4114-9ecb-f9d5c2f92671.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDExNjcwNTEsIm5iZiI6MTcwMTE2Njc1MSwicGF0aCI6Ii8zMjY0MjkyLzI4NTQ1OTcxMi1jZmExNzQ0ZS0zYTBkLTQxMTQtOWVjYi1mOWQ1YzJmOTI2NzEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTEyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzExMjhUMTAxOTExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWY4NzMwNDU4MTNhMDBmOGRlZTk1OTJhZjE1NDc4MmI2MjVhM2M1MzQ1NWNjMGVhMzQzZTMxZTNiNjhiNjM2NiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.yi1hx5YN8q7qcf43jpg_QS3xiVdeUp_tsLx4KJ3JK94)

Hi, it seems that you were not using tmux's mouse support, but using your terminal app's mouse support. (Because when you select the IP, your cursor stays at the shell's prompt.) In that case, your terminal app takes care the selection for you.

You can test with `set -g mouse on`. I Updated my comment.

**Comment by bland328** [comment]
The release notes for tmux 3.3 say `* Respond to OSC 4 queries`.  Does that mean, for example, that I should expect this to return palette entry #8 when run from within tmux 3.3a?

printf '\033]4;8;?\a'

My terminal is iTerm2, and this command works outside of tmux, but returns nothing from within.

Though I'm not clear if they are needed, I do have `allow-passthrough` turned on, and have terminal overrides `,xterm*:Tc` and `,xterm*:RGB`.

**Comment by souvikmishra** [comment]
Hi I was using tmux for a while now, recently i updated my mac to MacOS sonoma 14.1.2, suddenly each terminal session started opening at `/` instead of the usual `~` dir. It started messing with the integrated vsc terminal too. I was wondering if someone here might be able to point me in the right direction?

**Comment by XhstormR** [comment]
It has been more than a year, when will the next version be released? I'm looking forward to the hyperlink support in the next version.
